Marlin/src/module/temperature.cpp

```diff
@@ -54,32 +86,62 @@
   #endif
   #ifndef MAX31865_MOSI_PIN
     #define MAX31865_MOSI_PIN MOSI_PIN
-  #endif
-  #ifndef MAX31865_MISO_PIN
+  #endif  
+  #if !defined(MAX31865_MISO_PIN) && PIN_EXISTS(MAX6675_DO)
     #define MAX31865_MISO_PIN MAX6675_DO_PIN
   #endif
-  #ifndef MAX31865_SCK_PIN
+  #if !defined(MAX31865_SCK_PIN) && PIN_EXISTS(MAX6675_SCK)
     #define MAX31865_SCK_PIN  MAX6675_SCK_PIN
   #endif
   #if MAX6675_0_IS_MAX31865 && PIN_EXISTS(MAX31865_CS)
     #define HAS_MAX31865 1
     Adafruit_MAX31865 max31865_0 = Adafruit_MAX31865(MAX31865_CS_PIN
-      #if MAX31865_CS_PIN != MAX6675_SS_PIN
+      #if MAX31865_USES_SW_LIB
```

Comment:
And `MAX31865_CS_PIN != MAX6675_SS_PIN`, for me, have different meaning that `MAX31865_USES_SW_LIB `.

Reply:
What name do you want?  It is there to add DO_PIN and SCK_PIN and MOSI_PIN to the parameter list for the call to the class.  The class has two objects: SW obj which uses the Software SPI or HW obj which uses Hardware SPI.

Reply:
I changed the name from `MAX31865_USES_SW_LIB` to `MAX31865_USES_SW_SPI`